### PR TITLE
Compute IIS

### PR DESCRIPTION
**Pull request by dourouc05**
Following https://github.com/JuliaOpt/JuMP.jl/issues/1035: implement IIS support in a second package. 

This is just the first basic, baby step: retrieve data from the solver (only low-level API). It is only tested when the model is feasible, right now. I don't see any general use of the values of the duals nor the isolation thing, as they are very specific to Xpress (as far as I can tell). Or should I still implement them in this PR? (Maybe using optional parameters to `getfirstiis`?) Also, I ignore the fact that Xpress can compute several IISes: maybe there should be another function to return them all, or a variant that returns an IIS iterator? (Just thinking out loud, right now, but I'd like feedback on these ideas, especially on whether they should be implemented in this PR or not.) 

I tried to be as close as possible to https://github.com/JuliaOpt/CPLEX.jl/pull/233 (even copy-pasting the comments).

### CODE REVIEW

**Comment by dourouc05** [comment]
I guess that's pretty much it. I added tests for the C API, plus the MOI interface (basically, a copy-paste from CPLEX), plus tests. I had to remove everything that relates to SingleVariable, as it is not supported for Xpress. 

Right now, all the new tests are passing (I have many issues with the MOI tests not passing, most of them being that SingleVariable is not supported).

File: src/xprs_model.jl [hunk 1]
```diff
@@ -387,4 +387,77 @@ function setlogfile(model::Model, filename::String)
         throw(XpressError(model))
     end
     nothing
-end
\ No newline at end of file
+end
+
+mutable struct IISData
```
Comment (joaquimg) [inline src/xprs_model.jl]: I think the entire IIS thing should live in xprs_iis.jl, as you mentioned there are many ways to improve the interface.

**Comment by dourouc05** [comment]
There is no difference in SUPPORTED_CONSTRAINTS with, for instance, CPLEX, except order and `(LQOI.Linear, LQOI.IV)`, which only Xpress supports. (To ease comparison, maybe there should be guide lines about the order defined at LQOI or MOI level?)

However, two failures are related to start values: 

```
MathOptInterface.UnsupportedAttribute{MathOptInterface.VariablePrimalStart}: Attribute MathOptInterface.VariablePrimalStart() is not supported by the model.
```

I could not find the relevant functions to call in Xpress in their manual, except for "warm-starting is nice when you reoptimize". I guess it means the user cannot specify a solution? For now, I used the same solution as CPLEX: disable the corresponding tests. 

There are also a couple of warnings about MPB: 

```
running mathprog.jl ...
Testing linproginterface with solver Xpress.XpressSolver
Test Summary:                                                                                                         | Pass  Total
Testing LP interface with Xpress.XpressSolver(Base.Iterators.Pairs{Union{},Union{},Tuple{},NamedTuple{(),Tuple{}}}()) |   88     88
Done
┌ Warning: Element type Int64 does not have an infinite value. Note that this may artifically introduce ranged (two-sided) constraints. To avoid this, consider casting the problem data to Float64.
└ @ MathProgBase.HighLevelInterface C:\Users\kwsq4764\.julia\packages\MathProgBase\rVlFR\src\HighLevelInterface\HighLevelInterface.jl:9
┌ Warning: Element type Int64 does not have an infinite value. Note that this may artifically introduce ranged (two-sided) constraints. To avoid this, consider casting the problem data to Float64.
└ @ MathProgBase.HighLevelInterface C:\Users\kwsq4764\.julia\packages\MathProgBase\rVlFR\src\HighLevelInterface\HighLevelInterface.jl:9
┌ Warning: Element type Int64 does not have an infinite value. Note that this may artifically introduce ranged (two-sided) constraints. To avoid this, consider casting the problem data to Float64.
└ @ MathProgBase.HighLevelInterface C:\Users\kwsq4764\.julia\packages\MathProgBase\rVlFR\src\HighLevelInterface\HighLevelInterface.jl:9
┌ Warning: Element type Int64 does not have an infinite value. Note that this may artifically introduce ranged (two-sided) constraints. To avoid this, consider casting the problem data to Float64.
└ @ MathProgBase.HighLevelInterface C:\Users\kwsq4764\.julia\packages\MathProgBase\rVlFR\src\HighLevelInterface\HighLevelInterface.jl:9
┌ Warning: Element type Int64 does not have an infinite value. Note that this may artifically introduce ranged (two-sided) constraints. To avoid this, consider casting the problem data to Float64.
└ @ MathProgBase.HighLevelInterface C:\Users\kwsq4764\.julia\packages\MathProgBase\rVlFR\src\HighLevelInterface\HighLevelInterface.jl:9
┌ Warning: Element type Int64 does not have an infinite value. Note that this may artifically introduce ranged (two-sided) constraints. To avoid this, consider casting the problem data to Float64.
└ @ MathProgBase.HighLevelInterface C:\Users\kwsq4764\.julia\packages\MathProgBase\rVlFR\src\HighLevelInterface\HighLevelInterface.jl:9
Test Summary:                                                                                 | Pass  Total
Testing linprog and subfunctions with Xpress.XpressSolver(Base.Iterators.Pairs(:PRESOLVE=>0)) |   26     26
Test Summary:                                                                                                       | Pass  Total
Testing mixintprog with Xpress.XpressSolver(Base.Iterators.Pairs{Union{},Union{},Tuple{},NamedTuple{(),Tuple{}}}()) |    3      3
Test Summary:                                                                        | Pass  Total
Testing quadprog with Xpress.XpressSolver(Base.Iterators.Pairs(:BARGAPSTOP=>1.0e-9)) |    9      9
Test Summary:                                                                                                           | Pass  Total
Testing SOCP interface with Xpress.XpressSolver(Base.Iterators.Pairs{Union{},Union{},Tuple{},NamedTuple{(),Tuple{}}}()) |    3      3
Testing coniclineartest with solver Xpress.XpressSolver
Test Summary:                                                                                                                                    | Pass  Total
Testing linear problems through conic interface with Xpress.XpressSolver(Base.Iterators.Pairs{Union{},Union{},Tuple{},NamedTuple{(),Tuple{}}}()) |   24     24
Done
Testing conicSOCtest with solver Xpress.XpressSolver
Test Summary:                                                                                                                                 | Pass  Total
Testing SOC problems through conic interface with Xpress.XpressSolver(Base.Iterators.Pairs{Union{},Union{},Tuple{},NamedTuple{(),Tuple{}}}()) |   26     26
Done
```

Maybe there are still other MOI tests that do not pass, but the tests are just sluggish… (at least on this computer).

**Comment by joaquimg** [comment]
those warning are expected.
If CPLEX is disabling the tests we can do the same.

**Comment by dourouc05** [comment]
Should this be merged, now?

**Comment by joaquimg** [comment]
Sure, I plan to doublecheck and merge tomorrow.
Thanks for the bump.

**Pull request merged**